### PR TITLE
Fix error handling for constructors and Invoke

### PR DESCRIPTION
**Pull request by anuptalwalkar**
#43 
Provide, Invoke don't register type `error` which is recommended to be last parameter in the function return.

### CODE REVIEW

File: container.go [hunk 1]
```diff
@@ -64,15 +64,20 @@ func (c *Container) Invoke(t interface{}) error {
...
+			if values[count-1].Type() == _typeOfError {
+				if err, _ := values[count-1].Interface().(error); err != nil {
+					return errors.Wrapf(err, "Error executing the function %v", ctype)
+				}
+				count--
 			}
-			for _, v := range values {
-				switch v.Type().Kind() {
+		}
+		if count > 0 {
```
Comment (alsamylkin) [inline container.go]: combine two ifs?

File: container.go [hunk 2]
```diff
@@ -64,15 +64,20 @@ func (c *Container) Invoke(t interface{}) error {
...
 		values := cv.Call(args)
-
-		if len(values) > 0 {
-			if err, _ := values[len(values)-1].Interface().(error); err != nil {
-				return errors.Wrapf(err, "Error executing the function %v", ctype)
+		count := len(values)
+		if count > 0 {
+			if values[count-1].Type() == _typeOfError {
+				if err, _ := values[count-1].Interface().(error); err != nil {
+					return errors.Wrapf(err, "Error executing the function %v", ctype)
```
Comment (alsamylkin) [inline container.go]: lowercase errors, please..

File: internal/graph/node.go [hunk 3]
```diff
@@ -114,18 +114,22 @@ func (n *funcNode) value(g *Graph, objType reflect.Type) (reflect.Value, error)
...
-	// if last value is an error, it is returned as a separate argument, otherwise nil
-	err, _ = values[len(values)-1].Interface().(error)
-
-	for _, v := range values {
-		if objType == v.Type() {
-			return v, err
+	count := len(values)
+	if count > 0 {
+		// check if last argument is an error
+		if values[count-1].Type() == _typeOfError {
```
Comment (alsamylkin) [inline internal/graph/node.go]: isn't it already checked in Register()?

File: container_test.go [hunk 4]
```diff
@@ -319,6 +319,8 @@ func TestConstructorErrors(t *testing.T) {
 			var p1 *FlakyParent
 			err := c.Resolve(&p1)
 			if tt.wantErr != "" {
+				var registeredError *error
```
Comment (alsamylkin) [inline container_test.go]: What about invoke tests?

File: internal/graph/node.go (hunk 3 above)
Comment (anuptalwalkar) [inline internal/graph/node.go]: these are returned values from constructor execution, so need to make sure we are not adding them to dig.

**Pull request merged**